src/charm.py

```diff
@@ -112,7 +114,7 @@ def __init__(self, *args):
         self.legacy_db_admin_relation = DbProvides(self, admin=True)
         self.tls = PostgreSQLTLS(self, PEER_RELATION_NAME, [self.unit_pod_hostname])
 
-        self._cores = os.cpu_count()
+        self._cores = max(min(os.cpu_count(), 4), 2)
```

Comment:
Not strictly necessary, but limiting instances. We don't have to stop on upgrade since old pod goes away.